src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -169,7 +173,19 @@ protected function refreshUser(TokenInterface $token)
                 $token->setUser($refreshedUser);
 
                 if (null !== $this->logger) {
-                    $this->logger->debug('User was reloaded from a user provider.', array('username' => $refreshedUser->getUsername(), 'provider' => get_class($provider)));
+                    $impersonatorUsername = null;
+                    foreach ($token->getRoles() as $role) {
+                        if ($role instanceof SwitchUserRole) {
+                            $impersonatorUsername = $role->getSource()->getUsername();
+                            break;
+                        }
+                    }
+
+                    $this->logger->debug('User was reloaded from a user provider.', array(
+                        'provider' => get_class($provider),
+                        'username' => $refreshedUser->getUsername(),
+                        'impersonator_username' => $impersonatorUsername,
```

Comment:
I'd like to see this info only on switching user, so maybe `impersonator_username` could be ignored if is it `null`?

Reply:
agreed, let's not add it if no switch_user in place.

Reply:
Shouldnt it be impersonated_username?

Reply:
@ro0NL not sure https://symfony.com/doc/current/security/impersonating_user.html

Reply:
You're right :) i mixed up.

> In some cases you may need to get the object that represents the impersonating user rather than the impersonated user.

`impersonating_username` then?

cc @gharlan

Reply:
https://github.com/symfony/symfony-docs/pull/8320

Reply:
Right :) :+1: